tests/system/action/poll/test_vote.py

```diff
@@ -0,0 +1,382 @@
+        )
+        self.create_model(
+            "user/2", {"username": "test2", "is_present_in_meeting_ids": [113]}
+        )
+        self.update_model("user/1", {"is_present_in_meeting_ids": [113]})
+        self.create_model(
+            "poll/1",
+            {
+                "title": "my test poll",
+                "global_option_id": 11,
+                "global_no": True,
+                "global_yes": False,
+                "global_abstain": False,
+                "meeting_id": 113,
+                "entitled_group_ids": [1],
+            },
+        )
+        self.create_model("meeting/113", {"name": "my meeting"})
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [{"id": 1, "user_id": 1, "value": "N"}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        vote = self.get_model("vote/1")
+        assert vote.get("value") == "N"
+        assert vote.get("option_id") == 11
+        assert vote.get("weight") == "1.000000"
+        assert vote.get("meeting_id") == 113
+        assert vote.get("user_id") == 1
+        option = self.get_model("option/11")
+        assert option.get("vote_ids") == [1]
+        assert option.get("yes") == "0.000000"
+        assert option.get("no") == "1.000000"
+        assert option.get("abstain") == "0.000000"
+        user = self.get_model("user/1")
+        assert user.get("vote_$_ids") == ["113"]
+        assert user.get("vote_$113_ids") == [1]
+
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [{"id": 1, "user_id": 2, "value": "Y"}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_not_exists("vote/2")
+        option = self.get_model("option/11")
+        assert option.get("vote_ids") == [1]
+        user = self.get_model("user/1")
+        assert user.get("vote_$_ids") == ["113"]
+        assert user.get("vote_$113_ids") == [1]
+
+    def test_vote_schema_problems(self) -> None:
+        self.create_model("group/1", {"user_ids": [1]})
+        self.create_model(
+            "poll/1", {"title": "my test poll", "entitled_group_ids": [1]}
+        )
+        self.create_model("meeting/113", {"name": "my meeting"})
+        self.update_model("user/1", {"is_present_in_meeting_ids": [113]})
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [{"id": 1, "user_id": 1, "value": "X"}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 400)
+        assert (
+            "data.value must be valid by one of anyOf definition"
```

Comment:
This is not a good error to send to the client. Unfortunately, jsonschema isn't very helpful there. I would suggest that you don't try to mix the schematas like you currently do, but instead leave the schema of the action itself empty and first check the pollmethod of the poll and then manually check with the correct schema. This way, you don't have to use `anyOf` (hopefully) and can give reasonable error messages.

Reply:
Replaced AnyOf with code validation.